src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunction.cs

```diff
@@ -403,6 +403,23 @@ public virtual bool CheckTypes(CheckTypesContext context, TexlNode[] args, DType
             return CheckTypesCore(context, args, argTypes, errors, out returnType, out nodeToCoercedTypeMap);
         }
 
+        /// <summary>
+        /// Perform sub-expression type checking and produce a return type for the function declaration, this is only applicable for UDFs.
+        /// </summary>
+        public virtual bool CheckTypesOnDeclaration(CheckTypesContext context, TexlNode[] args, DType[] argTypes, DType returnType, IErrorContainer errors, out DType actualReturnType)
```

Comment:
These virtuals should not be here. They need to be in the UesrDefinedTexlFunction file.

Reply:
Moved them to a new like we discussed offline